src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
>If the user has not explicitly set Proxy value, whether the next http call goes via a proxy or not is decided by such an environment variable. 

That only means the value of the 'default proxy'.

in general, here are rules of semantics for HttpClientHandler and the `UseProxy` property vs. the `Proxy` property:

if (!UseProxy)
{
   // never use a proxy.  It doesn't matter if the user has set the `Proxy` property to a non-null value.
}
else
{
   if (Proxy != null)
   {
       // use a proxy and specially use a custom proxy using the IWebProxy interface
   }
   else
   {
      // use the 'default' system proxy.  This might be no proxy at all if the environment variable isn't set.  Otherwise,
      // a proxy will be used and it is based on the 'system' (or environment variable in this case) settings.
   }
}